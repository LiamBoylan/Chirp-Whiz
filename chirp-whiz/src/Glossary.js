import React, { useEffect, useState, useReducer } from 'react';
import AudioButton from './quizComponents/AudioButton';
import {
  Switch,
  Route,
  Link,
  useRouteMatch,
  useParams
} from "react-router-dom";
import 'bootstrap/dist/css/bootstrap.min.css';
import Button from 'react-bootstrap/Button';
import ButtonGroup from 'react-bootstrap/ButtonGroup';
import ListGroup from 'react-bootstrap/ListGroup';
import Card from 'react-bootstrap/Card';
import Image from 'react-bootstrap/Image';
import Container from 'react-bootstrap/Container';
import Col from 'react-bootstrap/Col';
import Row from 'react-bootstrap/Row';
import Carousel from 'react-bootstrap/Carousel';
import Form from 'react-bootstrap/Form';
import birdList from './birdList';



function Glossary () {
  const [state, setState] = useState({
    todos: birdList
  });
  const [birdNum, setBirdNum] = useState(0);
  const [inBirdPage, setInBirdPage] = useState(false);
  const [searchName, setSearchName] = useState('');
  

  useEffect(() => {
    window.scrollTo(0, 0);
  }, [birdNum])
  
  useEffect(() => {
    console.log(searchName);
  }, [searchName])

  function handleChange(e) {
    setSearchName(e.target.value);
  };

  return(
    <div>
      {state.todos.length > 0 &&
        <div>
          <Switch>
            <Route path={`${match.path}/:birdName`}>
              <Bird bird={state.todos[birdNum]}/>
              <Link to={match.path}>
                <Button variant="secondary">Return</Button>
              </Link>
            </Route>
            <Route path={match.path}>
              <div>
              <h1>Glossary</h1>
              <ButtonGroup vertical>
              <ListGroup variant="flush">
              {state.todos.map((todo, i) =>
                <ListGroup.Item>
                <Link to={`${match.url}/${todo.name}`} key={todo.id}>
                  <Button variant="outline-warning" style={{color: 'black'}} onClick={() => setBirdNum(i)}>
                    <p>{todo.name}</p>
                    <Image src={todo.image[0]} rounded style={{height: '300px'}} />
                  </Button>
                </Link>
                </ListGroup.Item>
              )}
              </ListGroup>
              </ButtonGroup>
              </div>
            </Route>
          </Switch>
        </div>
      }
    </div>
  );
}

export function Bird(props) {
  let { birdName } = useParams();

  return(
    <div>
      <Card className="text-center">
      <Card.Header><h2>{props.bird.name}</h2></Card.Header>
      <Card.Body>
      <Container>
      <Row>
      <Col>
      <Carousel>
        {props.bird.image.map((imag, i) =>
          <Carousel.Item>
            <Image src={imag} rounded style={{height: '360px'}} />
          </Carousel.Item>
        )}
      </Carousel>
      </Col>
      <Col>
      <AudioButton sound={props.bird.sound}/>
      <br />
      <h3>Scientific Name:</h3>
      <h5>{props.bird.scientificName}</h5><br />
      <h3>Bird Type:</h3>
      <h5>{props.bird.birdType}</h5><br />
      <h3>Habitat:</h3>
      <h5>{props.bird.habitat}</h5><br />
      </Col>
      </Row>
      </Container>
      </Card.Body>
      </Card>
    </div>
  );
}

export default Glossary;