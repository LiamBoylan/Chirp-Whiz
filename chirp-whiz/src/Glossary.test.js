import React from 'react';
import '@testing-library/jest-dom/extend-expect'
import { render, fireEvent } from '@testing-library/react'
import Glossary from './Glossary'

window.scrollTo = jest.fn();

// Integration tests

it('Glossary snapshot', () => {
    const container = render(<Glossary />)
    expect(container.firstChild).toMatchSnapshot();
    window.scrollTo.mockClear();
});

test("test if glossary renders correctly", () => {
    const {getByText} = render(<Glossary />)
    expect(getByText("Glossary")).toBeInTheDocument();
});

test("check out bird info page then return to glossary", () => {
    const {getByText} = render(<Glossary />)
    fireEvent.click(getByText('Evening grosbeak'));
    expect(getByText("Return")).toBeInTheDocument();
    fireEvent.click(getByText('Return'));
    expect(getByText("Glossary")).toBeInTheDocument();
});

const setup = () => {
  const utils = render(<Glossary />)
  const input = utils.getByLabelText('Bird Search:')
  return {
    input,
    ...utils,
  }
}

test('testing the search feature', () => {
  const { input } = setup()
  fireEvent.change(input, { target: { value: 'Even' } })
  expect(input.value).toBe('Even')
})