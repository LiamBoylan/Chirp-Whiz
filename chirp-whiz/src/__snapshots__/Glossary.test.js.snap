// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Home renders 1`] = `undefined`;
